client/checkout/woopay/direct-checkout/index.js

```diff
@@ -9,59 +9,119 @@ import { debounce } from 'lodash';
 	if ( isThirdPartyCookieEnabled ) {
 		if ( await WooPayDirectCheckout.isUserLoggedIn() ) {
 			WooPayDirectCheckout.maybePrefetchEncryptedSessionData();
-			WooPayDirectCheckout.redirectToWooPay( checkoutElements, true );
+			WooPayDirectCheckout.addRedirectToWooPayEventListener(
+				checkoutButtons,
+				true
+			);
 		}
 
 		return;
 	}
 
 	// Pass false to indicate we are not sure if the user is logged in or not.
-	WooPayDirectCheckout.redirectToWooPay( checkoutElements, false );
+	WooPayDirectCheckout.addRedirectToWooPayEventListener(
+		checkoutButtons,
+		false
+	);
+};
+
+/**
+ * Add an event listener to the mini cart checkout button.
+ */
+const addMiniCartEventListener = async () => {
+	const checkoutButton = WooPayDirectCheckout.getMiniCartProceedToCheckoutButton();
+	handleWooPayDirectCheckout( [ checkoutButton ] );
+};
+
+/**
+ * If the mini cart widget is available on the page, observe when the drawer element gets added to the DOM.
+ *
+ * As of today, no window events are triggered when the mini cart is opened or closed,
+ * nor there are attribute changes to the "open" button, so we have to rely on a MutationObserver
+ * attached to the `document.body`, which is where the mini cart drawer element is added.
+ */
+const maybeObserveMiniCart = () => {
+	// Check if the widget is available on the page.
+	if ( ! document.querySelector( '.wc-block-mini-cart' ) ) {
+		return;
+	}
+
+	// Create a MutationObserver to check when the mini cart drawer is added to the DOM.
+	const observer = new MutationObserver( ( mutations ) => {
+		for ( const mutation of mutations ) {
+			if ( mutation?.addedNodes?.length > 0 ) {
+				for ( const node of mutation.addedNodes ) {
+					// Check if the mini cart drawer parent selector was added to the DOM.
+					if (
+						node.nodeType === 1 &&
+						node.matches(
+							'.wc-block-components-drawer__screen-overlay'
+						)
+					) {
+						// Wait until the button is rendered and add the event listener to it.
+						waitForSelector(
+							WooPayDirectCheckout.redirectElements
+								.BLOCKS_MINI_CART_PROCEED_BUTTON,
+							addMiniCartEventListener
+						);
+						return;
+					}
+				}
+			}
+		}
+	} );
+
+	observer.observe( document.body, { childList: true } );
+};
+
+window.addEventListener( 'load', async () => {
+	if ( shouldSkipWooPay() ) {
+		return;
+	}
+
+	WooPayDirectCheckout.init();
+
+	isThirdPartyCookieEnabled = await WooPayDirectCheckout.isWooPayThirdPartyCookiesEnabled();
+
+	// If the mini cart is available, check when it's opened so we can add the event listener to the mini cart's checkout button.
+	maybeObserveMiniCart();
+
+	const checkoutButtons = WooPayDirectCheckout.getCheckoutButtonElements();
+	handleWooPayDirectCheckout( checkoutButtons );
 } );
 
 jQuery( ( $ ) => {
 	$( document.body ).on( 'updated_cart_totals', async () => {
-		if (
-			! WooPayDirectCheckout.isWooPayDirectCheckoutEnabled() ||
-			shouldSkipWooPay()
-		) {
+		if ( shouldSkipWooPay() ) {
```

Comment:
I agree.

Instead of checking for `shouldSkipWooPay`, I added the hooks in the `load` window event listener and after `isWooPayThirdPartyCookiesEnabled()` in [5ce2d84](https://github.com/Automattic/woocommerce-payments/pull/8795/commits/5ce2d84f3c3ee2514995758f8f1ff6e75451a7fe). LMK what you think.

@lovo-h – I tested this by adding a `console.log` to each callback and going to the blocks cart and adding, removing and setting the quantity of an item, however I wasn't able to hit the action hook when adding a new item from the suggestions list after the cart is emptied, am I testing this wrong? What's one way to hit the `experimental__woocommerce_blocks-cart-add-item`?

Reply:
> Instead of checking for `shouldSkipWooPay`, I added the hooks in the `load` window event listener and after `isWooPayThirdPartyCookiesEnabled()` in [5ce2d84](https://github.com/Automattic/woocommerce-payments/pull/8795/commits/5ce2d84f3c3ee2514995758f8f1ff6e75451a7fe). LMK what you think.

I think this approach looks cleaner. Good call.

>  I wasn't able to hit the action hook when adding a new item from the suggestions list after the cart is emptied, am I testing this wrong? What's one way to hit the `experimental__woocommerce_blocks-cart-add-item`?

Good catch.

I also wasn't able to trigger this callback either. It appears like it was removed in WooCommerce:
* [WooCommerce#43177](https://github.com/woocommerce/woocommerce/pull/43177)

If I recall correctly, this hook was added when wcBlocks was an independent plugin from WooCommerce. In wcBlocks, it looks like the [`experimental__woocommerce_blocks-cart-add-item` hook still exists](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/atomic/blocks/product-elements/add-to-cart/shared/add-to-cart-button.tsx#L157-L160).

One solution would be to replace the `experimental__woocommerce_blocks-cart-add-item` hook with `wc-blocks_added_to_cart`:
```javascript
document.body.addEventListener(
  'wc-blocks_added_to_cart',
  addItemCallback
);
```

However, this is an unrelated issue to this PR. So, we can create a separate GH issue.

Reply:
Created an issue. Feel free to look into this if you have the availability.

https://github.com/Automattic/woocommerce-payments/issues/8831